components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/processor/SAMLIDRequestProcessor.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.identity.query.saml.processor;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.opensaml.saml.saml2.core.AssertionIDRef;
+import org.opensaml.saml.saml2.core.AssertionIDRequest;
+import org.opensaml.saml.saml2.core.Issuer;
+import org.opensaml.saml.saml2.core.RequestAbstractType;
+import org.opensaml.saml.saml2.core.Response;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.model.SAMLSSOServiceProviderDO;
+import org.wso2.carbon.identity.query.saml.QueryResponseBuilder;
+import org.wso2.carbon.identity.query.saml.handler.SAMLAssertionFinder;
+import org.wso2.carbon.identity.query.saml.handler.SAMLAssertionFinderImpl;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is used to process AssertionIDRequest message
+ *
+ * @see AssertionIDRequest
+ */
+public class SAMLIDRequestProcessor implements SAMLQueryProcessor {
+
+    private final static Log log = LogFactory.getLog(SAMLIDRequestProcessor.class);
+
+    /**
+     * This process method is for requesting existing assertion from assertion store
+     *
+     * @param request AssertionIDRequest from requester
+     * @return Response Generated response message including assertions
+     */
+    public Response process(RequestAbstractType request) {
+        AssertionIDRequest assertion = (AssertionIDRequest) request;
+        String issuerFullName = getIssuer(request.getIssuer());
+        String issuer = MultitenantUtils.getTenantAwareUsername(issuerFullName);
```

Comment:
change the variable name to reflect it's true meaning, ie. something like tenantAwareIssuerName